openmc/deplete/integrator/si_celi.py

```diff
@@ -0,0 +1,177 @@
+"""The SI-CE/LI CFQ4 integrator."""
+
+import copy
+from collections.abc import Iterable
+
+from .cram import deplete
+from ..results import Results
+from ..abc import OperatorResult
+from .celi import _celi_f1, _celi_f2
+
+
+def si_celi(operator, timesteps, power=None, power_density=None,
+            print_out=True, m=10):
+    r"""Deplete using the SI-CE/LI CFQ4 algorithm.
+
+    Implements the Stochastic Implicit CE/LI Predictor-Corrector algorithm using
+    the [fourth order commutator-free integrator]_.
+
+    The CE/LI algorithm is mathematically defined as:
+
+    .. math:
+        y' = A(y, t) y(t)
+        A_p = A(y_n, t_n)
+        y_p = expm(A_p h) y_n
+        A_c = A(y_p, t_n)
+        A(t) = t/dt * A_c + (dt - t)/dt * A_p
```

Comment:
For the stochastic implicit versions of CE/LI and LE/QI, perhaps we ought to just omit the equations since they are not easily expressed in a few lines. Instead, you could just reference the relevant sections of Colin's thesis for further detail.